C++/3. Operators.md

```diff
@@ -47,13 +43,12 @@ int main() {
 7
 1
 12
-1
-1
+1.33333
 ```
 
-## Truncation
+## Types
 
-One major thing in C++ is truncation. As you may or may not have noticed, in the last example we had the **integer** 4 and the **integer** 3 dividing: `4 / 3`. The result of that was 1. The decimal points got truncated because the variable in which the result of that operation was stored was an integer. <!--If anyone find a way to fix this and make it show the decimal please tell me. I thought double or float would work but they don't too-->
+One problem that beginners may face when using in C++ is truncation. In the last example we had the **double** 4.0 and the **double** 3.0 dividing: `4.0 / 3.0`. The result of that was (1.33333...). `double` means double-precision, that is, when compared to a `float`. `double` and `float` are number types that can hold integers, decimals, infinity (inf), or not-a-number (nan), whereas integer types can only hold, well... *integers*. If you were to do `4 / 3`, this would perform integer division, and would return another integer, meaning that the would-be decimal places are lost.
```

Comment:
Wait no try a C++ program with a double. I tried it and it still truncated so I removed this point. I did the same thing you did first, but I changed it later.

Reply:
No no, try it yourself, you likely did this:
```c++
double half = 1 / 2;
```
And didn't get `0.5`, this is because 1 and 2 are integer literals, not floating-point values.

`1.0 / 2.0` is correct.

Reply:
o ok thanks for explaining. I did exactly that 😆 me dumb